### PR TITLE
Add toAnyContractKey and fromAnyContractKey

### PR DESCRIPTION
**Pull request by cocreature**
This is necessary to add exerciseByKey to DAML triggers.

See https://github.com/digital-asset/ghc/pull/44 for the corresponding PR in GHC.

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/unreleased.rst), if appropriate
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs [hunk 1]
```diff
@@ -651,7 +673,7 @@ convertBind env (name, x)
     --
     -- TODO(MH): The check is an approximation which will fail when users
     -- start the name of their own methods with, say, `_exercise`.
-    | any (`T.isPrefixOf` getOccText name) [ "$" <> prefix <> "_" <> method | prefix <- ["dm", "c"], method <- ["create", "fetch", "exercise", "toAnyTemplate", "fromAnyTemplate", "_templateTypeRep", "fetchByKey", "lookupByKey", "toAnyChoice", "fromAnyChoice"] ]
+    | any (`T.isPrefixOf` getOccText name) [ "$" <> prefix <> "_" <> method | prefix <- ["dm", "c"], method <- ["create", "fetch", "exercise", "toAnyTemplate", "fromAnyTemplate", "_templateTypeRep", "fetchByKey", "lookupByKey", "toAnyChoice", "fromAnyChoice", "_toAnyContractKey", "_fromAnyContractKey", "toAnyContractKey", "fromAnyContractKey"] ]
```
Comment (aherrmann-da) [inline compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs]: Why both the `_` prefixed and non-prefixed version? And if that's necessary, then maybe also for `_fromAnyChoice`, `_toAnyChoice`, and `templateTypeRep`?

Comment (cocreature) [reply compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs]: Both seems to be necessary. I’m a bit surprised why it doesn’t seem to be necessary for the AnyChoice methods but I haven’t investigated why it’s not an issue there.

Comment (hurryabit) [reply compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs]: I'm surprised we need the version without an `_` prefix here. To me, this feels like we're removing too much. Let's please find out what's going on here before we merge.

Comment (cocreature) [reply compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs]: Fixed, the one we don’t need is the one with the `_`, that is only added to the `TemplateKey` instance but not to the method in the `XInstance` typeclass. I messed this up in `ProposalDesugared` during development which made me think we need to filter both.

**Pull request merged**